python/cugraph/cugraph/gnn/feature_storage/feat_storage.py

```diff
@@ -0,0 +1,122 @@
+# Copyright (c) 2022-2023, NVIDIA CORPORATION.
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+from collections import defaultdict
+from typing import Any, Union
+import cudf
+import cupy as cp
+import numpy as np
+import pandas as pd
+from cugraph.utilities.utils import import_optional
+
+torch = import_optional("torch")
+
+
+class FeatureStore:
+    """The feature-store class used to store feature data for GNNS"""
+
+    def __init__(self, backend="numpy"):
+        self.fd = defaultdict(dict)
+        if backend not in ["numpy", "torch"]:
+            raise ValueError(
+                f"backend {backend} not supported. Supported backends are numpy, torch"
+            )
+        self.backend = backend
+
+    def add_data(self, feat_obj: Any, type_name: str, feat_name: str) -> None:
```

Comment:
Thanks for remembering the type annotations.

Should `feat_obj` be `Sequence` though?

Reply:
Agreed, we should make it a `Sequence` .

Reply:
Resolved by https://github.com/rapidsai/cugraph/pull/3139/commits/ae689fcd6609e3dcd946b7240c9e75ce6f5d7a1a